packages/app-mobile/components/HandleNoteQuery.ts

```diff
@@ -0,0 +1,41 @@
+import SearchEngineUtils from '@joplin/lib/services/searchengine/SearchEngineUtils';
+import SearchEngine from '@joplin/lib/services/searchengine/SearchEngine';
+import Note from '@joplin/lib/models/Note';
+
+// Returns notes from a search query and sets highlighted words
+// Be sure to use 'await' keyword or the function might not work properly
+// Eg. await HandleNoteQuery();
+
+const HandleNoteQuery = async (query: string, settings: any, dispatch: (action: Object)=> void): Promise<any[]> => {
```

Comment:
Pass only dbFtsEnabled (boolean), not the whole settings array. In general you should only pass what's strictly necessary to a function, which makes it easier to test it and use it.